mongoclient/db.go

```diff
@@ -0,0 +1,106 @@
+package mongoclient
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/netlify/netlify-commons/nconf"
+)
+
+const (
+	CollectionBlobs         = "blobs"
+	CollectionResellers     = "resellers"
+	CollectionUsers         = "users"
+	CollectionSubscriptions = "bb_subscriptions"
+	CollectionSites         = "projects"
+)
+
+type Auth struct {
+	Username string
+	Password string
+	Source   string
+}
+
+type Config struct {
+	AppName            string
+	TLS                *nconf.TLSConfig
+	Servers            []string
+	ReplSetName        string
+	ConnTimeout        time.Duration
+	Auth               *Auth
+	SecondaryPreferred bool
```

Comment:
If you're not specifying `split_words:"true"` then this is going to be kinda hackey in the env files.

Reply:
Yeah, so I intentionally didn't add the config file tags here because I wanted to separate the driver/API config from the config file parsing. I found each project has slightly different config requirements, along with test code, that combining the too was pretty messy. This allows the projects to manage their own config file formatting for the options uniquely.

Reply:
I agree that the mashup is kinda ugly (do you add `mapstructure` or `split_words` etc) ~ but this means that each person would have to re-implement this struct & add their flags. Personally, I like the idea of "add this config to your struct, then add the env var" setup. So you don't have a lot of mapping code (`libconfig.SecondaryPreferred = mycode.SecondaryPreferred` kinda things) & we get more consistent naming.